src/legacy/config/config-set.spec.ts

```diff
@@ -687,10 +688,10 @@ describe('resolvePath', () => {
 }) // resolvePath
 
 describe('_resolveTsConfig', () => {
-  let findConfig!: jest.SpyInstance<string | undefined>
+  let findConfig!: SpyInstance<(...args: any[]) => string | undefined>
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  let readConfig!: jest.SpyInstance<{ config?: any; error?: ts.Diagnostic }>
-  let parseConfig!: jest.SpyInstance<ts.ParsedCommandLine>
+  let readConfig!: SpyInstance<(...args: any[]) => { config?: any; error?: ts.Diagnostic }>
+  let parseConfig!: SpyInstance<(...args: any[]) => ts.ParsedCommandLine>
```

Comment:
```suggestion
  let readConfig!: SpyInstance<typeof ts.readConfigFile>
  let parseConfig!: SpyInstance<typeof ts.parseJsonConfigFileContent>
```

Seems like `jest.Mocked` would do better. Working on it (https://github.com/facebook/jest/pull/12727).

Interestingly `jest.mocked()` can’t be used as replacement. I was in doubt if both of them are needed. Thanks for motivation. I will push forward with `jest.Mocked`.